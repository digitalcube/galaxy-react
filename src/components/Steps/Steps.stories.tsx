import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Steps, StepsProps } from './Steps';

const meta: Meta = {
  title: 'Components/Steps',
  component: Steps,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<
  Omit<StepsProps, 'steps'> & {
    stepLength: number;
  }
> = ({ stepLength, ...args }) => {
  const steps = [...Array(stepLength)].map((_d, i) => ({
    name: `Step name ${i + 1}`,
    description: 'Step Descriptions',
  }));
  return <Steps {...args} steps={steps} />;
};

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  className: '',
  stepLength: 3,
  currentStep: 1,
  title: 'Title',
  footer: 'Footer',
};
