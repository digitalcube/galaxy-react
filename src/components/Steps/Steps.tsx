import React, { FC, ReactNode } from 'react';
export type StepDetail = {
  className?: string;
  name: ReactNode;
  description?: ReactNode;
  title?: ReactNode;
};
export type StepsProps = {
  currentStep: number;
  title?: ReactNode;
  steps: StepDetail[];
  footer?: ReactNode;
  className?: string;
};
export const Steps: FC<StepsProps> = ({
  currentStep,
  className,
  steps,
  title,
  footer,
}) => {
  return (
    <div
      className={`px-3 py-3 rounded steps fade show${
        className ? ` ${className}` : ''
      }`}
      role="alert"
    >
      {title ? (
        <header className="steps-header mb-4 d-flex align-items-center justify-content-between">
          <h2 className="mb-0 font-weight-bold steps-title">{title}</h2>
        </header>
      ) : null}
      <div className="container">
        <div className="row">
          {steps.map((step, i) => (
            <div
              className={`col-sm mx-1 text-center${
                currentStep === i + 1 ? ` is-current` : ''
              }${step.className ? ` ${step.className}` : ''}`}
              key={`step-${i}`}
            >
              {step.title ? (
                step.title
              ) : (
                <div className={`step-num font-weight-bold`}>{i + 1}</div>
              )}
              <div className="mt-2 font-weight-bold step-name">{step.name}</div>
              {step.description ? (
                <div className="mt-2 step-description">{step.description}</div>
              ) : null}
            </div>
          ))}
        </div>
      </div>
      {footer ? (
        <div className="steps-footer mt-3 px-3 align-items-center font-weight-bold">
          {footer}
        </div>
      ) : null}
    </div>
  );
};
