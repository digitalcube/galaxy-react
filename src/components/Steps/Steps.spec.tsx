import { render } from '@testing-library/react';
import React from 'react';
import { Default as Steps } from './Steps.stories';

describe('Steps', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(<Steps {...(Steps.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot (with error massage)', () => {
    const { asFragment } = render(
      <Steps
        {...(Steps.args as any)}
        errorMessage="We donâ€™t recognize that email"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
