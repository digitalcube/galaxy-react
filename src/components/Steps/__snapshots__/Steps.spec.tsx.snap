// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps to match snapshot (with error massage) 1`] = `
<DocumentFragment>
  <div
    class="px-3 py-3 rounded steps fade show"
    role="alert"
  >
    <header
      class="steps-header mb-4 d-flex align-items-center justify-content-between"
    >
      <h2
        class="mb-0 font-weight-bold steps-title"
      >
        Title
      </h2>
    </header>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm mx-1 text-center is-current"
        >
          <div
            class="step-num font-weight-bold"
          >
            1
          </div>
          <div
            class="mt-2 font-weight-bold step-name"
          >
            Step name 1
          </div>
          <div
            class="mt-2 step-description"
          >
            Step Descriptions
          </div>
        </div>
        <div
          class="col-sm mx-1 text-center"
        >
          <div
            class="step-num font-weight-bold"
          >
            2
          </div>
          <div
            class="mt-2 font-weight-bold step-name"
          >
            Step name 2
          </div>
          <div
            class="mt-2 step-description"
          >
            Step Descriptions
          </div>
        </div>
        <div
          class="col-sm mx-1 text-center"
        >
          <div
            class="step-num font-weight-bold"
          >
            3
          </div>
          <div
            class="mt-2 font-weight-bold step-name"
          >
            Step name 3
          </div>
          <div
            class="mt-2 step-description"
          >
            Step Descriptions
          </div>
        </div>
      </div>
    </div>
    <div
      class="steps-footer mt-3 px-3 align-items-center font-weight-bold"
    >
      Footer
    </div>
  </div>
</DocumentFragment>
`;

exports[`Steps to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-3 py-3 rounded steps fade show"
    role="alert"
  >
    <header
      class="steps-header mb-4 d-flex align-items-center justify-content-between"
    >
      <h2
        class="mb-0 font-weight-bold steps-title"
      >
        Title
      </h2>
    </header>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-sm mx-1 text-center is-current"
        >
          <div
            class="step-num font-weight-bold"
          >
            1
          </div>
          <div
            class="mt-2 font-weight-bold step-name"
          >
            Step name 1
          </div>
          <div
            class="mt-2 step-description"
          >
            Step Descriptions
          </div>
        </div>
        <div
          class="col-sm mx-1 text-center"
        >
          <div
            class="step-num font-weight-bold"
          >
            2
          </div>
          <div
            class="mt-2 font-weight-bold step-name"
          >
            Step name 2
          </div>
          <div
            class="mt-2 step-description"
          >
            Step Descriptions
          </div>
        </div>
        <div
          class="col-sm mx-1 text-center"
        >
          <div
            class="step-num font-weight-bold"
          >
            3
          </div>
          <div
            class="mt-2 font-weight-bold step-name"
          >
            Step name 3
          </div>
          <div
            class="mt-2 step-description"
          >
            Step Descriptions
          </div>
        </div>
      </div>
    </div>
    <div
      class="steps-footer mt-3 px-3 align-items-center font-weight-bold"
    >
      Footer
    </div>
  </div>
</DocumentFragment>
`;
